### ISSUE TITLE
API issue

### ISSUE THREAD

**Issue by rawadrifai** [faermanj/breakless#2]
Hey - so I watched your full video on youtube. What was the issue with the token not working? I was guessing that it's expired.

Also what's the difference between sessionToken and idToken?

**Comment by faermanj** [comment]
Yeah i was actually using the wrong token if i recall correctly but now with AWS Amplify it's easier to let the library handle that.

**Comment by rawadrifai** [comment]
Amplify changed everything. I'm in love.
Quick question, do you have an example for doing the federated identity flow?
What I don't understand well is:
1- If we are using the default cognito auth flow (code grant), should there be a backend server that exchanges the auth code for a token set? Right now, we sign in and receive tokens, no?

2- If we authenticate with a federated identity like google, we have to give it the callbacks, correct? Can we do it with no callbacks? I assume yes, and that's token auth flow.. Is that good practice though? If we give it a callback, then that's a server we manage in order to exchange the code for tokens in the back channel. How do we pass the tokens back to the client?

**Comment by faermanj** [comment]
1- No, the Cognito service actually handles that for you, no need to manage servers yourself.

2- The "callbacks" are actually URLs called in each step of the authentication flow, depending on the protocol, but the Cognito service and the Amplify library handles that in a way that makes it more convenient to plug in your you application without coupling to the protocols of each authentication provider. There is even a default UI so you can start with a functional and secure user registration and authentication flow.

**Comment by kyeung2** [comment]
Hi Like the other users I also watched your video and got to the end where there's that issue with API Gateway.

Maybe it would be good to add a comment on this repo's README? Letting users know the current state, give them some direction? Like the preferred integration is with the Amplify framework?

And add a link to this repo on the [video](https://www.youtube.com/watch?v=TowcW1aTDqE)?

Anyway, thanks for the video, I learnt a lot.
